### PR TITLE
Add ConvRNN-T Encoder

### PR DESCRIPTION
**Pull request by yfyeung**
Add ConvRNN-T Encoder
Paper: ConvRNN-T: Convolutional Augmented Recurrent Neural Network Transducers for Streaming Speech Recognition
https://assets.amazon.science/b3/30/8c96a61247f5970baae7198abe38/convrnn-t-convolutional-augmented-recurrent-neural-network-transducers-for-streaming-speech-recognition.pdf

### CODE REVIEW

File: egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py [hunk 1]
```diff
@@ -0,0 +1,1029 @@
...
+        if not rd_channels:
+            rd_channels = self.make_divisible(channels * rd_ratio, rd_divisor, round_limit=0.)
+        self.fc1 = nn.Conv1d(channels, rd_channels, kernel_size=1, bias=bias)
+        self.act = nn.ReLU(inplace=True)
+        self.fc2 = nn.Conv1d(rd_channels, channels, kernel_size=1, bias=bias)
+        self.gate = nn.Sigmoid()
+
+    def forward(self, x):
+        # x: (N, C, T)
+        x_se = x.mean(2, keepdim=True)
```
Comment (kobenaxie) [inline egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:273]: `x.mean(2, keepdim=True)` here needs to consider padding ?

File: egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py [hunk 2]
```diff
@@ -0,0 +1,1029 @@
...
+        layer_dropout (float): layer-dropout rate.
+        causal (bool): Whether to use causal convolution in ConvRNNT encoder
+            layer.
+    """
+
+    def __init__(
+        self,
+        num_features: int,
+        d_model: int = 512,
+        LocalCNNEncoder_kernel_size: int = 5,
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:39]: Please use a consistent style for naming variables, e.g.,
```
local_cnn_encoder_kernel_size
```

File: egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py [hunk 3]
```diff
@@ -0,0 +1,1029 @@
...
+
+    def __init__(
+        self,
+        in_channels: int,
+        out_channels: int,
+    ) -> None:
+        """
+        Args:
+          in_channels:
+            Number of channels in. The input shape is (N, T, in_channels).
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:740]: The implementation does not match the comment. Please fix it.

File: egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py [hunk 4]
```diff
@@ -0,0 +1,1029 @@
+#!/usr/bin/env python3
```
Comment (csukuangfj) [inline egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:1]: Please create a new directory, eg, pruned_transducer_stateless7, and add training script for it.

**Review by csukuangfj — CHANGES_REQUESTED** [review]

File: egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py (hunk 1 above)
Comment (yfyeung) [reply egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:273]: You are right, I will fix it.

Comment (yfyeung) [reply egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:273]: You are right, I will fix it.

Comment (kobenaxie) [reply egs/librispeech/ASR/pruned_transducer_stateless5/ConvRNN-T.py:273]: Another question, if mean over all time steps T, is it suitable for streaming ?
